pkg/apis/serving/v1alpha1/kfservice_types.go

```diff
@@ -42,6 +41,29 @@ type ModelSpec struct {
 	XGBoost    *XGBoostSpec    `json:"xgboost,omitempty"`
 	SKLearn    *SKLearnSpec    `json:"sklearn,omitempty"`
 	PyTorch    *PyTorchSpec    `json:"pytorch,omitempty"`
+	Explain    *ExplainerSpec  `json:"explain,omitempty"`
```

Comment:
I think this is at the right level. We will probably want to push the 1-of predictors down a level under `predict`

Reply:
sorry for missing yesterday's meeting. Conceptually, I am having hard time convincing why would Explainer Spec be at the same level as specs for 'Model Types' e.g. PyTorch, or Custom Model?

Reply:
@animeshsingh The core proposals are in this [doc](https://docs.google.com/document/d/1V-ATKWEuV4-uqkc6CqMsgmkTFt-2Sf1wBofpBE1bqBI/edit) discussed at the meeting. The main one for level is that Explainers can load saved models (some need to be fit on the training data) and they can be canaried (as they may fail). Eventually we would put the Predictor specs as @ellis-bigelow mentioned in a level further down to separate them more clearly - see my earlier commit.

Reply:
@ellis-bigelow Are you suggesting to push the other predictors down a level now? I thought the idea was to not break the YAML specs at present?

Reply:
s/Explain/Explainer
or
s/ExplainerSpec/ExplainSpec

Reply:
I think `explain` is matching the future `predict` (see first comment). It could indeed also be `explainer` and `predictor`.

Reply:
Updated

Reply:
I think it is worth considering using v1alpha2 if we do not want to break existing yaml, patching to existing v1alpha1 might be confusing to user. We can do conversion from v1alpha1 to v1alpha2 in mutation webhook if we really want to back support v1alpha1, I believe knative does this too between v1alpha1 and v1beta1 .

Reply:
I have added a comment to the spec to make it clear the current location of `explain`. I think the advantage of keeping it at v1alpha1 is that we can get a working example and leave the v1alpha2 changes to a separate piece of work. Doing them together will be a bigger PR. 

I'm happy either way. Just want to keep momentum.